packages/lesswrong/components/form-components/FormComponentMultiSelect.tsx

```diff
@@ -19,36 +25,53 @@ const styles = (theme: ThemeType): JssStyles => ({
   }
 })
 
-const FormComponentMultiSelect = ({ value, classes, placeholder, separator, options, path, updateCurrentValues }) => {
+type MultiselectOption = {
+  value: string,
+  label: string
+}
+
+const FormComponentMultiSelect = ({ value, label, placeholder, separator, options, path, updateCurrentValues, classes }: {
+  value: Array<string>,
+  label?: string,
+  placeholder?: string,
+  separator?: string,
+  options: Array<MultiselectOption>,
+  path: string,
+  updateCurrentValues<T extends {}>(values: T): void,
+  classes: ClassesType
+}) => {
   
-  return <Select
-    className={classes.root}
-    value={value}
-    input={<OutlinedInput labelWidth={0} />}
-    onChange={e => {
-      // MUI documentation says e.target.value is always an array: https://mui.com/components/selects/#multiple-select
-      // @ts-ignore
-      updateCurrentValues({
-        [path]: e.target.value
-      })
-    }}
-    multiple
-    displayEmpty
-    renderValue={(selected: Array<string>) => {
-      if (selected.length === 0) {
-        return <em className={classes.placeholder}>{placeholder}</em>
-      }
-      // if any options are selected, display them separated by commas
-      return selected.map(s => options.find(option => option.value === s)?.label).join(separator || ', ')
-    }}
-    {...!options.length ? {disabled: true} : {}}>
-      {options.map(option => {
-        return <MenuItem key={option.value} value={option.value}>
-          <Checkbox checked={value.some(v => v === option.value)} />
-          <ListItemText primary={option.label} />
-        </MenuItem>
-      })}
-  </Select>
+  return <FormControl className={classes.root}>
+    {label && <FormLabel className={classes.label}>{label}</FormLabel>}
+    <Select
+      className={classes.select}
+      value={value}
+      input={<OutlinedInput labelWidth={0} />}
+      onChange={e => {
+        // MUI documentation says e.target.value is always an array: https://mui.com/components/selects/#multiple-select
+        // @ts-ignore
+        updateCurrentValues({
+          [path]: e.target.value
+        })
```

Comment:
I'd rather do an explicit type cast (`e.target.value as ...`) rather than a blanket `// @ts-ignore`, which seems like it could potentially let through a wider range of unexpected runtime bugs

Reply:
(just realized that this isn't a new piece of code 😓 - would be nice to fix but not that important)

Reply:
Hmm ts seems unhappy with the type cast since it thinks `e.target.value` is a `string` - I'm not sure what the proper ts way to handle this is?